include/zephyr/mem_mgmt/mem_attr.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MEM_ATTR_H_
+#define ZEPHYR_INCLUDE_MEM_ATTR_H_
+
+/**
+ * @brief Memory-Attr Interface
+ * @defgroup memory_attr_interface Memory-Attr Interface
+ * @ingroup mem_mgmt
+ * @{
+ */
+
+#include <stddef.h>
+#include <zephyr/types.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @cond INTERNAL_HIDDEN */
+
+#define _DT_MEM_ATTR	zephyr_memory_attr
```

Comment:
Yes there is a reason. I have always used names beginning with underscore to mark something that is reserved for the current file and usually not exported outside, so this is a perfect example of this use case (the underscore is basically saying "there is nothing to see, move along"). It is also not inlined because I want this to stands out when reading the header file for self-documentation.

Reply:
Changed with `__MEM_ATTR`